### DISCUSSION TITLE
Any plans for translation?

### DISCUSSION THREAD

**Discussion by Bamowen** [TagStudioDev/TagStudio#438] [General]
**EDIT 06.09**: This is now discussed in https://github.com/TagStudioDev/TagStudio/issues/446. Closing this discussion.

Hi! 
I don't have much programming skill, but I'd love to contribute more.
I can clearly help with translation if needed. Maybe we can setup something like [Crowdin](https://crowdin.com), which is what Mastodon is using for instance and would streamline all languages.

Any thoughts?

**Comment by CyanVoxel** [comment]
Hello! Translation is definitely planned, but nothing's been started with it as of yet. Now that 9.4 is out this would probably be a great time to get that ball rolling! There's an issue open for it (#83) but not specific implementations have been discussed there yet.

**Comment by Nginearing** [comment]
I'd definitely be willing to help if needed, I'm not exactly "fluent" in the languages I've been studying for the last few years but my knowledge can definitely help, since there's only so many strings to be localised.

**Comment by Bamowen** [comment]
Alright! 
Is there any existing list of every string in the app, possibly with their keys?
We can put that in a json to start if it doesn't exist :)

**Comment by Tyrannicodin** [comment]
I'm going to try doing crowdin integration, I plan on using json files located at `/tagstudio/resources/translations/`

**Reply by Bamowen** [reply]
Are you creating the project in Crowdin? I can do that as well if needed and see who needs File addition access.
By default, every user joining is a Translator and can suggest translation. Only certain roles can add files.

Source for different roles: https://support.crowdin.com/modifying-project-participants-roles/

**Reply by Tyrannicodin** [reply]
I created the project, if you send me your username I'll add you and give you permissions to manage it, I'm on discord if you'd rather send it to me there.

**Reply by Bamowen** [reply]
My username is Bamowen on Crowdin

I've not joined the discord yet

**Reply by Tyrannicodin** [reply]
I've invited you, for some reason I couldn't give higher permissions than translator, so I'll try and upgrade your permissions after you join.

**Reply by Nginearing** [reply]
Is there not a link that you can give for anyone that wants to join? As there are a lot of languages people will want to contribute, it will be painstaking to add everyone 1 by 1.

**Reply by Tyrannicodin** [reply]
I can: https://crowdin.com/project/tag-studio/invite?h=09592c47c760ee353c64e1f8f6053faf2202369

**Reply by Nginearing** [reply]
> I can: https://crowdin.com/project/tag-studio/invite?h=09592c47c760ee353c64e1f8f6053faf2202369
Great.
Does crowdin also have GitHub integration? as in it can automatically create pull requests for languages?

**Reply by Tyrannicodin** [reply]
Yes it can, I'll need to talk to Cyan about that once it's ready

**Reply by Nginearing** [reply]
> Yes it can, I'll need to talk to Cyan about that once it's ready

That sounds wonderful, I guess we either have a separate localisation branch or we have a localisation folder that is only included in the source code not in releases, to reduce bloat bloat, the actual software would use the github api to check for any new languages and then it would allow the user to install the languages from there.

**Reply by Bamowen** [reply]
Do we need to exclude them from releases?
I'm looking over at [Mastodon repo](https://github.com/mastodon/mastodon) and they have it directly at app/javascript/mastodon/locales/en.json

**Reply by Nginearing** [reply]
@CyanVoxel What do you think about using Crowdin? Tyrannicodin is getting to grips with it, would you want to have full control over the translation on Crowdin or just create the actual Crowdin project and then just add someone like tyrannicodin or Bamowen as a maintainer (I think they are called managers on Crowdin) Crowdin honestly seems like a smooth way to localise (translate) the UI, there is progress management also on Crowdin to make it easier for managers to keep track of the localisation process.

**Reply by CyanVoxel** [reply]
> @CyanVoxel What do you think about using Crowdin? Tyrannicodin is getting to grips with it, would you want to have full control over the translation on Crowdin or just create the actual Crowdin project and then just add someone like tyrannicodin or Bamowen as a maintainer (I think they are called managers on Crowdin) Crowdin honestly seems like a smooth way to localise (translate) the UI, there is progress management also on Crowdin to make it easier for managers to keep track of the localisation process.

I'm currently looking into Crowdin now

**Reply by Nginearing** [reply]
> I'm currently looking into Crowdin now

I've already got a feature request written up, specifically detailing what it entails for the program, as the other translation feature request is not properly written out nicely. I'll post the feature request when you get back to us (and if you're up for it of course).

**Reply by Nginearing** [reply]
You can now join the discussion in #446.
Please no longer discuss on this page.

**Comment by Tyrannicodin** [comment]
Looks like Qt uses custom translation files translated using [Qt Linguist](https://doc.qt.io/qt-5/qtlinguist-index.html) so we'll have to use that I think

**Comment by Bamowen** [comment]
For completeness' sake, this is being discussed in https://github.com/TagStudioDev/TagStudio/issues/446

**Reply by Nginearing** [reply]
> For completeness' sake, this is being discussed in #446

Sorry, I forgot to put this, Mentioned that I was going to create one, but didn't link it.

**Reply by Nginearing** [reply]
I'd recommend changing the first post of this discussion to link directly to it, so people don't get confused, Please close this discussion too.

**Comment by Nginearing** [comment]
Hi all,

You are now able to translate the TagStudio user interface into your language of choice by clicking [here](https://hosted.weblate.org/engage/tagstudio/).

How the translation process is going so far.... ⬇️⬇️⬇️  - as of the 24th of September 2024
![multi-auto](https://github.com/user-attachments/assets/8525b049-7bd9-48bc-a262-9fb9ecfd98b2)

If you are still referring to this discussion for correspondence about translation support, please do not respond in this discussion, instead you should look at #446 for all future correspondence about this feature, where you can also ask any questions you may have.

And of course, make sure to join our [Discord server](https://discord.com/invite/hRNnVKhF2G), where we have a thread just for communicating about the Weblate translations!
